Analysis/Producer.py

```diff
@@ -117,19 +150,27 @@ def createHistograms(df_dict, var):
             anaskimmer.df = anaskimmer.df.Filter(f"X_mass=={args.mass}")
         anaskimmer.skimAnatuple()
         dataframes[sample] = anaskimmer.df
-
     defineWeights(dataframes)
+
     all_histograms = {}
+    all_sums = {}
     vars = args.vars.split(',')
+    all_sums = createSums(dataframes)
     for var in vars:
         hists = createHistograms(dataframes, var)
         all_histograms[var] = hists
 
+    hists_to_plot = {}
     for var in vars:
```

Comment:
To simplify `GetValue` procedure define:
```python
def GetValues(collection):
    for key, value in collection.items():
        if isinstance(value, dict):
            GetValues(value)
        else:
            collection[key] = value.GetValue()
```
and then call it before entering in the loops:
```
GetValues(all_histograms)
GetValues(all_sums)
```